emir/estimators/kernels/gaussian_kernel.py

```diff
@@ -86,30 +86,35 @@ def __init__(self, args, zc_dim, zd_dim, **kwargs):
         self.K = args.cond_modes
         self.logC = torch.tensor([-self.d / 2 * np.log(2 * np.pi)]).to(args.device)
 
-        self.mu = FF(args, zc_dim, self.ff_hidden_dim, self.K * self.d)
-        self.logvar = FF(args, zc_dim, self.ff_hidden_dim, self.K * self.d)
-
-        self.weight = FF(args, zc_dim, self.ff_hidden_dim, self.K)
-        self.tri = None
+        out_ff_dim = self.K * (2 * self.d + 1) # mu: d, logvar: d, w: 1
         if args.cov_off_diagonal == "var":
-            self.tri = FF(args, zc_dim, self.ff_hidden_dim, self.K * self.d**2)
+            out_ff_dim += self.K * self.d**2
+            self.tri = True
+        else:
+            self.tri = False
+
+        self.ff = FF(args, zc_dim, self.ff_hidden_dim, out_ff_dim)
```

Comment:
Ok, i need to dive into this, but we a hidden dim of self.ff_hidden_dim reduces the amount of parameter for each value predicted (before we had multiple FF, each having a hidden dim of self.ff_hidden_dim)